dive_core/data_core.h

```diff
@@ -30,6 +30,73 @@
 namespace Dive
 {
 
+#define CHECK_AND_TRACK_STATE_1(stats_enum, state)                   \
```

Comment:
i feel that it would be clearer if we separate these changes into separate trace_stats.h/cpp. data_core.h/cpp becomes too large with those changes.